### PR TITLE
Make v0.14.6 PS release

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Prep PR for making the v0.14.6 release

---

**Checklist:**

- [ ] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
Now that #4217 has been merged, should release guide be updated to remove the instructions specifying that?

**Comment by JordanMartinez** [comment]
CI is failing because we build the package set to verify no breaking changes were introduced in a PR. Unfortunately, it means the [`metadata` package](https://github.com/purescript/purescript/runs/5253780293?check_suite_focus=true#step:8:245) causes the build to fail. I'm looking into it.

**Comment by JordanMartinez** [comment]
To fix CI, it looks like we need to make an empty commit in [spacchetti/purescript-metadata](https://github.com/spacchetti/purescript-metadata/commits/master) and add a tag for it. Since that affects package sets, I'm not sure if that triggers anything else. @f-f Could you clarify what that will do?

**Comment by JordanMartinez** [comment]
I need one approval before I can merge this and make the next release.

File: CHANGELOG.md
Comment (natefaubion) [inline CHANGELOG.md:2598]: Whats all this about?

File: npm-package/package.json
Comment (natefaubion) [inline npm-package/package.json:49]: Missing newline?

File: CHANGELOG.md
Comment (JordanMartinez) [reply CHANGELOG.md:2598]: Ah... Markdown formatting. It tries to make the table easier to read by adding all that aligning whitespace automatically. I'll fix that.

Comment (JordanMartinez) [reply CHANGELOG.md]: Fixed.

File: npm-package/package.json
Comment (JordanMartinez) [reply npm-package/package.json]: Also fixed.

**Comment by f-f** [comment]
> Could you clarify what that will do?

The `metadata` package is a giant hack so that we can keep track of purs' version inside package sets. The reason why this hack exists is that we can't change package sets' format due to keeping compatibility with psc-package, and getting rid of this hack is one of the reasons why I'm restructuring Spago's config format.

I can cut a new release of that so you don't need to depend on a fork

**Comment by JordanMartinez** [comment]
@f-f I'd appreciate the new release. Thanks!

**Comment by f-f** [comment]
@JordanMartinez done!

**Comment by JordanMartinez** [comment]
Thanks!

**Comment by JordanMartinez** [comment]
Can I get an approval here?

**Comment by thomashoneyman** [comment]
@JordanMartinez I'm almost done with #4233; would you mind delaying this until Friday so this has a chance to go in first? If not, I can try to work it in to a 0.14.7 release instead.

**Comment by JordanMartinez** [comment]
@thomashoneyman Fine by me! But if it's delayed beyond Friday, then I will release this (so long as I get an approval).

**Comment by thomashoneyman** [comment]
Just waiting on reviews at this point, but the PR is complete.

**Comment by JordanMartinez** [comment]
I've updated this PR to account for the `purs publish` PR (#4233). Once CI builds, this just needs an approval, and then I can release it.

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**
